### PR TITLE
feat: add parser for @attr

### LINKED ISSUE #702: [🚀 esl-core]: ability to pass parser to the @attr definition
As an ESL consumer/contributor, I want to be able to define an attribute parser to parse and validate the value accessed from the DOM attribute. 

**Example**
```typescript
class MyElement extends HTMLELement{
    @attr({ parser: (val) => +val }) public value: number;
} 

...
expect(typeof $el.value).toBe('number');
```

### PR DESCRIPTION
**Pull request by Anna-MariiaPetryk**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

Closes #702

### CODE REVIEW

File: src/modules/esl-base-element/decorators/attr.ts [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
+type AttrDescriptor<T> = {
   /** HTML attribute name. Uses kebab-cased variable name by default */
   name?: string;
   /** Create getter only */
   readonly?: boolean;
   /** Use data-* attribute */
   dataAttr?: boolean;
   /** Default property value. Used if no attribute is present on the element. Empty string by default. */
-  defaultValue?: string | boolean | null;
+  defaultValue?: T ;
```
Comment (dshovchko) [inline src/modules/esl-base-element/decorators/attr.ts:16]:
```suggestion
  defaultValue?: T;
```

File: src/modules/esl-base-element/decorators/attr.ts [hunk 2]
```diff
@@ -38,10 +46,12 @@ const buildAttrName =
  * Maps string type property.
  * @param config - mapping configuration. See {@link AttrDescriptor}
  */
-export const attr = (config: AttrDescriptor = {}): PropertyDecorator => {
-  config = Object.assign({defaultValue: ''}, config);
+
+export const attr = <T = string>(config: AttrDescriptor<T> = {}): PropertyDecorator => {
+  config = Object.assign({defaultValue: '', serializer : {parse: identity, serialize: identity}}, config);
```
Comment (dshovchko) [inline src/modules/esl-base-element/decorators/attr.ts:51]:
```suggestion
  config = Object.assign({defaultValue: '', serializer: {parse: identity, serialize: identity}}, config);
```

File: src/modules/esl-base-element/decorators/serializers.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+
+import {ESLMediaQuery} from '../../all';
```
Comment (dshovchko) [inline src/modules/esl-base-element/decorators/serializers.ts:2]:
```suggestion
import {ESLMediaQuery} from '../../esl-media-query/core';
```

File: src/modules/esl-base-element/decorators/serializers.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import type {IMediaQueryCondition} from '../../esl-media-query/core/conditions/media-query-base';
+
+export type Serializer<T> = {
+  parse: (value: string) => T;
+  serialize: (value: T) => string | boolean | null;
+};
+
+export const  toNumber: Serializer<number> = {
```
Comment (dshovchko) [inline src/modules/esl-base-element/decorators/serializers.ts:10]:
```suggestion
export const toNumber: Serializer<number> = {
```

File: src/modules/esl-base-element/decorators/serializers.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  parse: (val: string): number => +val,
+  serialize: String,
```
Comment (dshovchko) [inline src/modules/esl-base-element/decorators/serializers.ts:12]:
```suggestion
  serialize: String
```

File: src/modules/esl-base-element/decorators/serializers.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+};
+
+export const  toMediaQueryCondition: Serializer<IMediaQueryCondition> = {
```
Comment (dshovchko) [inline src/modules/esl-base-element/decorators/serializers.ts:15]:
```suggestion
export const toMediaQueryCondition: Serializer<IMediaQueryCondition> = {
```

File: src/modules/esl-base-element/decorators/serializers.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  parse: ESLMediaQuery.for,
+  serialize: String,
```
Comment (dshovchko) [inline src/modules/esl-base-element/decorators/serializers.ts:17]:
```suggestion
  serialize: String
```

File: src/modules/esl-base-element/decorators/attr.ts (hunk 1 above)
Comment (Anna-MariiaPetryk) [reply src/modules/esl-base-element/decorators/attr.ts:16]: thanks for your attention, I’ll fix it up

**Comment by ala-n** [comment]
Closed as a change merged indirectly in #1041 with the latest requirements update

**Pull request closed without merging**